packages/@react-spectrum/combobox/src/ComboBox.tsx

```diff
@@ -276,6 +276,9 @@ const ComboBoxInput = React.forwardRef(function ComboBoxInput(props: ComboBoxInp
     }
 
     lastInputValue.current = inputValue;
+    return () => {
```

Comment:
This will run every time `[isLoading, showLoading, inputValue]` changes, which means we may delay certain timeouts.

Instead, this cleanup should be in its own `useEffect`, which runs on unmount only.
```
  useEffect(() => {
    return () => {
      clearTimeout(timeout.current);
      timeout.current = null;
    };
  }, []);
```

The addition of the timeout.current = null is needed to compensate for React StrictMode, otherwise when the second mount happens, the timer won't be set again

Reply:
Ah, that's a good point. Thanks for the follow-up!